### PR TITLE
More updates for DI types philosophy

### PR DESCRIPTION
**Pull request by weaverryan**
Hi guys!

This is another big (sorry) PR for updating things for using type-hinting for injection. I only have a few directories remaining to review!

FYI, the following directories I'm not planning on reviewing, at least not right away (so, these are up for grabs!)
- best_practices
- bundles
- components
- console
- create_framework
- reference

Thanks!

### CODE REVIEW

File: security/access_denied_handler.rst [hunk 1]
```diff
@@ -65,39 +30,43 @@ generally return a custom response).
...
-            http://symfony.com/schema/dic/services/services-1.0.xsd">
+    .. code-block:: xml
 
-            <services>
-                <service id="app.security.access_denied_handler"
-                        class="AppBundle\Security\AccessDeniedHandler" />
-            </services>
-        </container>
+        <config>
+          <firewall name="main">
```
Comment (javiereguiluz) [inline security/access_denied_handler.rst]: This cde uses 2-white space indentation. Should it be changed to 4?

File: security/api_key_authentication.rst [hunk 2]
```diff
@@ -389,40 +316,42 @@ using the ``simple_preauth`` and ``provider`` keys respectively:
...
         // app/config/security.php
 
-        // ..
+        // ...
+        use AppBundle\Security\ApiKeyAuthenticator;
+        use AppBundle\Security\ApiKeyUserProvider;
 
         $container->loadFromExtension('security', array(
+            'providers' => array(
+                'api_key_user_provider'  => array(
```
Comment (javiereguiluz) [inline security/api_key_authentication.rst]: There are two white spaces before the arrow operator here -> `'api_key_user_provider'  => array(`

File: security/api_key_authentication.rst [hunk 3]
```diff
@@ -682,65 +594,12 @@ current URL is before creating the token in ``createToken()``::
             // set the only URL where we should look for auth information
             // and only return the token if we're at that URL
             $targetUrl = '/login/check';
-            if (!$this->httpUtils->checkRequestPath($request, $targetUrl)) {
+            if ($request->getPathInfo() != $targetUrl)
```
Comment (javiereguiluz) [inline security/api_key_authentication.rst]: Can we use a strict checking here and replace `!=` by `!==`?

File: service_container/alias_private.rst [hunk 4]
```diff
@@ -43,29 +46,30 @@ public (i.e. private):
...
+        use AppBundle\Service\Foo;
 
-        $container->register('foo', Foo::class)
+        $container->register(Foo::class)
             ->setPublic(false);
 
-What makes private services special is that, since the container knows that the
-service will never be requested from outside, it can optimize whether and how it
-is instanciated. This increases the container's performance.
+Private services are special because it allows the container to optimize whether
```
Comment (javiereguiluz) [inline service_container/alias_private.rst]: `because it allows the container [...]` -> `because they allow the container [...]` ?

**Review by javiereguiluz — APPROVED** [review]
👍  Ryan, thanks for another great PR. And you removed double the lines that you changed/added, so this is fantastic!

**Comment by weaverryan** [comment]
> And you removed double the lines that you changed/added, so this is fantastic!

I didn't even realize! That's GREAT!

Thanks for the review - I know these are a pain, but you're helping me continue to move quickly and get this done. Merged this into the 3.3 branch

**Pull request closed without merging**